eos-tech-support/eos-write-live-image

```diff
@@ -0,0 +1,284 @@
+
+BOOT_ZIP="${OS_IMAGE%.img.gz}.boot.zip"
+check_exists "$BOOT_ZIP" "bootloader bundle"
+check_exists "$BOOT_ZIP.asc" "bootloader bundle signature"
+
+check_exists "$WINDOWS_TOOL" "Windows tool"
+
+echo
+echo "Partitioning $DEVICE..."
+
+find_by_type() {
+    echo "$1" | grep -i "type=$2" | cut -d' ' -f1
+}
+
+if $MBR; then
+    (
+        echo "1 : type=7"
+    ) | sfdisk --label dos $DEVICE
+    udevadm settle
+    partprobe $DEVICE
+    PARTMAP=$(sfdisk --dump $DEVICE)
+    DEVICE_IMAGES=$(find_by_type "$PARTMAP" "7")
+else
+    # https://en.wikipedia.org/wiki/GUID_Partition_Table#Partition_type_GUIDs
+    PARTITION_SYSTEM_GUID="c12a7328-f81f-11d2-ba4b-00a0c93ec93b"
+    PARTITION_BIOS_BOOT_GUID="21686148-6449-6E6F-744E-656564454649"
+    PARTITION_BASIC_DATA_GUID="ebd0a0a2-b9e5-4433-87c0-68b6b72699c7"
+
+    # We want the data partition, "eoslive", to occupy all the space on the disk
+    # after the UEFI and BIOS boot partitions. But, we also want it to be numbered
+    # first: apparently Windows will only mount the partition numbered first,
+    # regardless of where it is on the disk.
+    (
+        echo "2 : start=2048, size=62MiB, type=$PARTITION_SYSTEM_GUID"
+        echo "3 : size=1MiB, type=$PARTITION_BIOS_BOOT_GUID"
+        echo "1 : name=eoslive, type=$PARTITION_BASIC_DATA_GUID"
+    ) | sfdisk --label gpt $DEVICE
+    udevadm settle
+    partprobe $DEVICE
+    PARTMAP=$(sfdisk --dump $DEVICE)
+    DEVICE_IMAGES=$(find_by_type "$PARTMAP" "$PARTITION_BASIC_DATA_GUID")
+    DEVICE_EFI=$(find_by_type "$PARTMAP" "$PARTITION_SYSTEM_GUID")
+    DEVICE_BIOS=$(find_by_type "$PARTMAP" "$PARTITION_BIOS_BOOT_GUID")
+fi
+
+echo
+echo "Creating partitions"
+if ! $MBR; then
+    mkfs.vfat -n efi "${DEVICE_EFI}"
+fi
+$MKFS_IMAGES $MKFS_ARGS eoslive "${DEVICE_IMAGES}"
+
+# Give udisks a chance to notice the new partitions
+partprobe
+udevadm settle
+
+udisks_mount() {
+    udisksctl mount -b "$1" --no-user-interaction >&2
+    MOUNT_POINT=$(udisksctl info -b "$1" | grep "MountPoints:" | cut -f2 -d':')
+    test -d "$MOUNTPOINT"
+    echo $MOUNT_POINT
+}
+
+udisks_unmount() {
+    udisksctl unmount -b "$1" --no-user-interaction >&2
+}
+
+if ! $MBR; then
+    MOUNT_EFI=$(udisks_mount "$DEVICE_EFI")
+    unzip -d "${MOUNT_EFI}" "${BOOT_ZIP}" "EFI/*"
+    udisks_unmount "$DEVICE_EFI"
+fi
+
+MOUNT_IMAGES=$(udisks_mount "$DEVICE_IMAGES")
+MOUNT_IMAGES_ENDLESS="${MOUNT_IMAGES}/endless"
+mkdir "$MOUNT_IMAGES_ENDLESS"
+unzip -d "${MOUNT_IMAGES_ENDLESS}" "${BOOT_ZIP}" "grub/*"
+
+if ! $WRITABLE; then
+    cp "${BOOT_ZIP}" "${BOOT_ZIP}.asc" "${EXTRACTED_SIGNATURE}" \
+        "$MOUNT_IMAGES_ENDLESS/"
+    echo "$(basename "${OS_IMAGE%.gz}")" > "${MOUNT_IMAGES_ENDLESS}/live"
+    cp "$WINDOWS_TOOL" "$MOUNT_IMAGES/"
+fi
+
+"$EOS_WRITE_IMAGE" "$OS_IMAGE" "-" > "${MOUNT_IMAGES_ENDLESS}/endless.img"
+udisks_unmount "$DEVICE_IMAGES"
+
+if $MBR; then
+    unzip -p "${BOOT_ZIP}" "ntfs/boot.img" | dd of="${DEVICE}" bs=446 count=1
```

Comment:
Some comments here about the offsets would be nice as well as ones for settle/sync